File: 30 days of code JAVA/day18.java
import java.io.*;
import java.util.*;

public class Solution {
    // Write your code here.
  LinkedList s=new LinkedList();


    LinkedList q= new LinkedList();

void pushCharacter(char c)
{
    s.push(c);

}

char popCharacter()
{   

   return (char)s.pop();

}


void enqueueCharacter(char c)
{
    q.add(c);
}

char dequeueCharacter()
{
    
    return (char)q.remove();
}



    

    public static void main(String[] args) {

File: 30daysOFcode_Scala/day 18.scala
import scala.collection.mutable.ListBuffer
class Solution {
  //Write your code here
    var lb = new ListBuffer[Char];
    var ch = 'a';
    def outputListBuffer() = {
        println(lb)
    }
    def pushCharacter(c: Char) = {
        lb += c;
    }
    def enqueueCharacter(c: Char) = {
        //lb.insert(0, c);
    }
    def popCharacter() : Char = {
        ch = lb.last;
        if ( lb.size < 2 ) {
            return ch;
        }
        try {
            ch = lb.remove(lb.size-1);
        } catch {
            case _: Throwable => 
        }
        ch
    }
    def dequeueCharacter() : Char = {
        ch = ' ';
        try {
            ch = lb.remove(0);
        } catch {
            case _: Throwable => 
        }
        ch
    }
}
